### PR TITLE
fix partition prune

### PR DESCRIPTION
**Pull request by nautaa**
wrong case:
```
create table t (a DateTime, b UInt32) engine=BaseStorage partition by toYear(a)
insert into t values ('2020-01-01:08:00:00',1),('2021-01-01:08:00:00',1)
TensorBase :) select * from t where toYear(a)<>2020
SELECT *
FROM t
WHERE toYear(a) != 2020

Query id: d986e108-ea6a-42dc-8d18-3f261fa6e516

Ok.

0 rows in set. Elapsed: 6.991 sec.
```
also fix a related test case in `integ_test`.

### CODE REVIEW

File: crates/tests_integ/tests/sanity_checks.rs [hunk 1]
```diff
@@ -819,17 +819,17 @@ async fn tests_integ_select_remote_function() -> errors::Result<()> {
     insert.commit().await?;
 
     drop(insert);
-    let dates = [
-        Utc.ymd(2010, 1, 1).and_hms(0, 0, 0),
-        Utc.ymd(2011, 2, 28).and_hms(0, 0, 0),
-        Utc.ymd(2012, 2, 29).and_hms(0, 0, 0),
+    let dates = vec![
```
Comment (jinmingjian) [inline crates/tests_integ/tests/sanity_checks.rs:822]: why you change a testcase for remote functions? why not add a new testcase?

File: crates/tests_integ/tests/sanity_checks.rs [hunk 2]
```diff
@@ -1146,6 +1146,26 @@ async fn tests_integ_partition_prune() -> errors::Result<()> {
...
     }
 
+    conn.execute(format!("drop table if exists test2_tab"))
+        .await?;
+    conn.execute(format!(
+        "create table test2_tab (a UInt32, b UInt32) engine=BaseStorage partition by a"
+    ))
+    .await?;
+
+    conn.execute(format!("insert into test1_tab values(1,1),(2,2)"))
```
Comment (jinmingjian) [inline crates/tests_integ/tests/sanity_checks.rs:1156]: why ```create table test2_tab``` but ```insert into test1_tab```?

File: crates/tests_integ/tests/sanity_checks.rs [hunk 3]
```diff
@@ -1146,6 +1146,26 @@ async fn tests_integ_partition_prune() -> errors::Result<()> {
...
+        .await?;
+    {
+        let sql = "select * from test2_tab where toYear(a)<>1";
```
Comment (jinmingjian) [inline crates/tests_integ/tests/sanity_checks.rs:1159]: OMG, ```toYear(a)<>1```, is this test really passed?!  :smiley:

**Review by jinmingjian — COMMENTED** [review]
you truly find a problem, but you may not solve this problem.

**Comment by jinmingjian** [comment]
@nautaa you find two problems here in fact:
1. what I said about datetime handling problem in part expr calculation. you can skip this(about lightjit, a little hard)
2. new ```fill_copainfos_int_by_ptk_range``` has bugs but the pruning result is right.

File: crates/tests_integ/tests/sanity_checks.rs (hunk 1 above)
Comment (nautaa) [reply crates/tests_integ/tests/sanity_checks.rs:822]: @jinmingjian  this test case isn't correct and has an impact on this PR.

Comment (jinmingjian) [reply crates/tests_integ/tests/sanity_checks.rs:822]: ok. In this case, it is better to make a change for this testcase and then add a new one

**Comment by jinmingjian** [comment]
@nautaa your PR is a great catch for null partition logic handling! Thanks!

**Pull request merged**